### DISCUSSION TITLE
What is the purpose of "mmu" and "mtu" in the vision library?

### DISCUSSION THREAD

**Discussion by Sarayin1001** [Xilinx/Vitis_Libraries#142] [Q&A]
I'm having trouble understanding what MTU and MMU is for in the Vision Library, specifically for the Kalman filter function.  I can see a couple spots where these are defined, and there's an assert in xf_kalmanfilter.hpp, but I can't find any other lines that use it.

The documentation says they're used as multipliers for time and measurement updates, but what are they multiplying?  Why do they need to be greater than 1 or less than N_STATE?

**Comment by mahadurkarm** [comment]
MTU & MMU are used as PROC_TU & PROC_MU respectively by child functions in the xf_kalmanfilter.hpp. MTU & MMU are configurable parameter which are used to tune the performance & resource utilization.  if it is 1 then design will need lesser resources & give lesser performance. To improve performance based on the resource budget, you can increase value of these configurable parameter up to N_STATE.
